src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java

```diff
@@ -89,15 +92,18 @@
     private JsonParser parser;
     private String currentTypeStructure;
     private CompilerContext context = null;
+    private String projectDirectory;
 
     @Override
     public void setCompilerContext(CompilerContext ctx) {
         context = ctx;
+        CompilerOptions options = CompilerOptions.getInstance(ctx);
+        projectDirectory = options.get(CompilerOptionName.PROJECT_DIR);
+        this.dlog = new DataMapperDiagnosticLog(ctx);
```

Comment:
This will change to `this.dlog = DataMapperDiagnosticLog.getInstance(ctx);` with the change suggested below.

Reply:
Fixed in [3749d63](https://github.com/ballerina-platform/module-ballerinax-datamapper/pull/36/commits/3749d6315a93bed2c289bd683bbf40f8d541089a)